lib/mayaUsd/fileio/shading/shadingModeExporterContext.cpp

```diff
@@ -300,12 +300,20 @@ UsdMayaShadingModeExportContext::GetAssignments() const
         MDagPath dagPath = allDagPaths[instanceNumber];
         TF_VERIFY(dagPath.instanceNumber() == instanceNumber);
 #endif
-
         MFnDagNode dagNode(dagPath, &status);
         if (!status) {
             continue;
         }
 
+        ret.hasAnyAssignment = true;
+
+        // Note: the connections will be to the mesh, but the selection often
+        //       contains the transform above the mesh or a higher-up ancestor,
+        //       so also check the ancestors of the mesh.
+        for (MDagPath walkUpPath(dagPath); walkUpPath.length() > 0; walkUpPath.pop())
+            if (GetExportArgs().fullObjectList.hasItem(walkUpPath))
+                ret.hasAnyAssignmentInSelection = true;
+
```

Comment:
To support the logic of determining if a material should be exported, we need to know if it is used in the scene and if it used by something that was selected. This is what the new code above does.